File: admin-web/src/lib/iraqGovernorates.ts
// iraqGovernorates — Note #32. Fixed list of Iraq's 18 governorates, used to
// turn the beneficiary case form's free-text "City" field into a structured
// dropdown ("Governorate" per the client's note) instead of open text. No
// canonical governorate/city/neighborhood table exists anywhere in the
// schema, so this is a static list rather than an API-fetched one — the
// underlying `city` DB column/API field is unchanged, only its label and
// input widget change.
export type Governorate = { value: string; en: string; ar: string }

export const IRAQ_GOVERNORATES: Governorate[] = [
  { value: 'baghdad', en: 'Baghdad', ar: 'بغداد' },
  { value: 'basra', en: 'Basra', ar: 'البصرة' },
  { value: 'nineveh', en: 'Nineveh', ar: 'نينوى' },
  { value: 'erbil', en: 'Erbil', ar: 'أربيل' },
  { value: 'sulaymaniyah', en: 'Sulaymaniyah', ar: 'السليمانية' },
  { value: 'duhok', en: 'Duhok', ar: 'دهوك' },
  { value: 'kirkuk', en: 'Kirkuk', ar: 'كركوك' },
  { value: 'najaf', en: 'Najaf', ar: 'النجف' },
  { value: 'karbala', en: 'Karbala', ar: 'كربلاء' },
  { value: 'anbar', en: 'Anbar', ar: 'الأنبار' },
  { value: 'diyala', en: 'Diyala', ar: 'ديالى' },
  { value: 'wasit', en: 'Wasit', ar: 'واسط' },
  { value: 'saladin', en: 'Saladin', ar: 'صلاح الدين' },
  { value: 'babil', en: 'Babil', ar: 'بابل' },
  { value: 'qadisiyyah', en: 'Al-Qadisiyyah', ar: 'القادسية' },
  { value: 'muthanna', en: 'Al-Muthanna', ar: 'المثنى' },
  { value: 'dhi_qar', en: 'Dhi Qar', ar: 'ذي قار' },
  { value: 'maysan', en: 'Maysan', ar: 'ميسان' },
]

export function governorateLabel(value: string | null | undefined, locale: string | undefined): string {
  if (!value) return ''
  const g = IRAQ_GOVERNORATES.find((x) => x.value === value)
  if (!g) return value
  return locale === 'ar' || locale === 'ckb' || locale === 'kmr' ? g.ar : g.en
}

File: humanitarian/lib/data/iraq_governorates.dart
/// Iraq's 18 governorates, for the grantor registration form's governorate
/// dropdown. Each entry's translation key is the English name itself — add
/// the Arabic/Kurdish translations to app_translations.dart under the same
/// key to localize the dropdown.
const List<String> iraqGovernorates = [
  'Baghdad',
  'Basra',
  'Nineveh',
  'Erbil',
  'Sulaymaniyah',
  'Duhok',
  'Kirkuk',
  'Anbar',
  'Najaf',
  'Karbala',
  'Wasit',
  'Babil',
  'Diyala',
  'Salah al-Din',
  'Dhi Qar',
  'Maysan',
  'Muthanna',
  'Al-Qadisiyyah',
];

/// Resolves a stored city string to its governorate KEY, case- and
/// whitespace-insensitively, or '' when it is not one of the eighteen.
///
/// WHY THIS EXISTS
/// The keys are capitalised — 'Duhok' — because they are also the English
/// labels. But `city` is a free-text field on several forms, prefilled from a
/// `city_user` preference that itself came from typing, so what is actually
/// stored is whatever the volunteer wrote: `duhok`, `Duhok `, `DUHOK`. A card
/// then called `.tr` on that and got the string back unchanged, so an Arabic
/// screen read "duhok" beside «تم الإرسال» and «أربيل».
///
/// Matching case-insensitively against a CLOSED list of eighteen names is a
/// normalisation, not a guess: either the text is one of them or it is not,
/// and anything else — a village, a district, a typo — returns '' and is left
/// exactly as the person wrote it. That distinction is the whole point; a
/// looser match would start rewriting people's own words.
String? governorateKeyFor(String? city) {
  final value = (city ?? '').trim();
  if (value.isEmpty) return null;
  for (final name in iraqGovernorates) {
    if (name.toLowerCase() == value.toLowerCase()) return name;
  }
  return null;
}
